tests/subsys/suit/unit/suit_decrypt_filter/src/main.c

```diff
@@ -0,0 +1,545 @@
+ * which is in format |nonce (12 bytes)|tag (16 bytes)|ciphertext|
+ */
+
+static const uint8_t ciphertext_direct[] = {
+	/* tag (16 bytes) */
+	0x4d, 0x21, 0x30, 0xb7, 0xce, 0x8a, 0xd6, 0x00, 0xe4, 0x04, 0xbb, 0x32,
+	0x72, 0x7a, 0xbb, 0x7c,
+	/* ciphertext */
+	0xf0, 0x72, 0xdb, 0x63, 0x03, 0xdd, 0x24, 0x69,
+	0xd4, 0xbf, 0xd7, 0xa0, 0xec, 0xfa, 0x66, 0x58, 0x95, 0x2b, 0xc1, 0xc2,
+	0x9d, 0x82, 0x02, 0x1a, 0xd7, 0x5b, 0xc0, 0x01, 0xce, 0x0b, 0x79, 0x53,
+	0xe7, 0xdb, 0x0d, 0x35, 0xab, 0xef, 0x81, 0xc8, 0x68, 0xc5, 0xa7, 0x22,
+	0x90, 0xea, 0xd0, 0x7f, 0x36, 0xed, 0x14, 0xbe, 0x30, 0xf2, 0x81, 0x56,
+	0x7e, 0x2e, 0x5f, 0xd8, 0x7c,
+};
+
+
+static const uint8_t iv_direct[] = {
+	0x60, 0x90, 0x6d, 0xb2, 0xfe, 0xc3, 0xc8, 0x5a, 0xf0, 0x28, 0xb1, 0xb6,
+};
+
+static const suit_manifest_class_id_t sample_class_id = {
+	{0x5b, 0x46, 0x9f, 0xd1, 0x90, 0xee, 0x53, 0x9c, 0xa3, 0x18, 0x68, 0x1b, 0x03, 0x69, 0x5e,
+	 0x36}
+};
+
+static const uint8_t cek_key_id_cbor[] = {
+		0x1A, 0x40, 0x02, 0x20, 0x00,
+	};
+
+struct suit_decrypt_filter_tests_fixture {
+	struct stream_sink dec_sink;
+	struct stream_sink ram_sink;
+	size_t decrypted_output_length;
+	psa_status_t aead_update_status;
+	psa_status_t aead_verify_status;
+	suit_plat_err_t ram_sink_write_status;
+
+	/* ram_sink interfaces call counters. Not really 'fixtures', but something
+	 *  we verify as a side effect of test run.
+	 */
+	int ram_sink_write_call_cnt;
+	int ram_sink_erase_call_cnt;
+	int ram_sink_release_call_cnt;
+};
+
+static struct suit_decrypt_filter_tests_fixture tests_fixture;
+
+static const uint8_t aad[] = {
+	"sample aad"
+};
+
+/* Custom fake functions implementation - for returning values by reference. */
+static suit_plat_err_t custom_suit_plat_decode_key_id(struct zcbor_string *key_id,
+									uint32_t *integer_key_id)
+{
+	(void)key_id;
+
+	*integer_key_id = KEY_ID_FWENC_APPLICATION_GEN1;
+	return SUIT_PLAT_SUCCESS;
+}
+
+static psa_status_t custom_psa_aead_update(psa_aead_operation_t *operation,
+									const uint8_t *input,
+									size_t input_length,
+									uint8_t *output,
+									size_t output_size,
+									size_t *output_length)
+{
+	(void)operation;
+	(void)input;
+	(void)input_length;
+	(void)output;
+	(void)output_size;
+
+	*output_length = tests_fixture.decrypted_output_length;
+
+	return tests_fixture.aead_update_status;
+}
+
+static psa_status_t custom_psa_aead_verify(psa_aead_operation_t *operation,
+									uint8_t *plaintext,
+									size_t plaintext_size,
+									size_t *plaintext_length,
+									const uint8_t *tag,
+									size_t tag_length)
+{
+	(void)operation;
+	(void)plaintext;
+	(void)plaintext_size;
+	(void)tag;
+	(void)tag_length;
+
+	*plaintext_length = tests_fixture.decrypted_output_length;
+
+	return tests_fixture.aead_verify_status;
+}
+
+/* dummy interface functions for the decrypted data output sink (fixture->ram_sink). */
+static suit_plat_err_t release(void *ctx)
```

Comment:
I would rather create `FAKE_VALUE_FUNCTION` defining all this functions and use fff mechanisms to count the call counts